include/beman/scope/scope.hpp

```diff
@@ -113,7 +133,276 @@ make_unique_resource_checked(R&& r, const S& invalid, D&& d) noexcept(noexcept(
+};
+
+//======
+
+// --- Specializations for no releaser
+
+template <scope_exit_function ScopeExitFunc>
+class [[nodiscard]] scope_guard<ScopeExitFunc, void, exception_during_constuction_behaviour::invoke_exit_func> {
+    ScopeExitFunc m_exit_func;
+
+  public:
+    template <typename T>
+    explicit constexpr scope_guard(T&& exit_func) noexcept(std::is_nothrow_constructible_v<ScopeExitFunc>)
+        requires(!std::is_same_v<std::remove_cvref<T>, scope_guard>)
+    try : m_exit_func(std::forward<T>(exit_func)) {
+    } catch (...) {
+        exit_func();
+
+        throw;
+    }
+
+    scope_guard(const scope_guard&)            = delete;
+    scope_guard(scope_guard&&)                 = delete;
+    scope_guard& operator=(const scope_guard&) = delete;
+    scope_guard& operator=(scope_guard&&)      = delete;
+
+    constexpr ~scope_guard() noexcept(noexcept(m_exit_func())) { m_exit_func(); }
+};
+
+//======
+
+template <scope_exit_function ScopeExitFunc>
+class [[nodiscard]] scope_guard<ScopeExitFunc, void, exception_during_constuction_behaviour::dont_invoke_exit_func> {
+    ScopeExitFunc m_exit_func;
+
+  public:
+    template <typename T>
+    explicit constexpr scope_guard(T&& exit_func) noexcept(std::is_nothrow_constructible_v<ScopeExitFunc>)
+        requires(!std::is_same_v<std::remove_cvref<T>, scope_guard>)
+        : m_exit_func(std::forward<T>(exit_func)) {}
+
+    scope_guard(const scope_guard&)            = delete;
+    scope_guard(scope_guard&&)                 = delete;
+    scope_guard& operator=(const scope_guard&) = delete;
+    scope_guard& operator=(scope_guard&&)      = delete;
+
+    constexpr ~scope_guard() noexcept(noexcept(m_exit_func())) { m_exit_func(); }
+};
+
+//==================================================================================================
+
+// --- Deduction guides ---
+
+template <typename ExitFunc,
+          typename InvokeChecker,
+          exception_during_constuction_behaviour ecdb = exception_during_constuction_behaviour::invoke_exit_func>
+    requires(scope_exit_function<ExitFunc> && (scope_function_invoke_check<InvokeChecker>))
+scope_guard(ExitFunc&&, InvokeChecker&&) -> scope_guard<std::decay_t<ExitFunc>, std::decay_t<InvokeChecker>, ecdb>;
+
+template <typename ExitFunc,
+          typename InvokeChecker                      = void,
+          exception_during_constuction_behaviour ecdb = exception_during_constuction_behaviour::invoke_exit_func>
+    requires(scope_exit_function<ExitFunc> &&
+             (scope_function_invoke_check<InvokeChecker> || std::is_void_v<InvokeChecker>))
+scope_guard(ExitFunc&&) -> scope_guard<std::decay_t<ExitFunc>, InvokeChecker, ecdb>;
+
+//==================================================================================================
+
+class Releaser {
+  public:
+    bool operator()() const { return m_can_invoke; }
+
+    void release() { m_can_invoke = false; }
+
+  private:
+    bool m_can_invoke = true;
```

Comment:
```suggestion
    bool can_invoke_ = true;
```
This is a global comment.  The _style_ of the standard is to use underscore after the member name.  It would be helpful if our implementation would follow that style so that we can trivially cut and paste from the code into the paper wording.